packages/experimental/navigation/otel/hooks/useTrace.ts

```diff
@@ -0,0 +1,39 @@
+import {trace, Tracer, TracerProvider} from '@opentelemetry/api';
+import {MutableRefObject, useEffect, useRef} from 'react';
+
+const TRACER_DEFAULT = {
+  name: 'default',
+  version: '1.0',
+};
+
+interface ConfigArgs {
+  name?: string;
+  version?: string;
+}
+
+type TracerRef = MutableRefObject<Tracer | null>;
+
+const useTrace = (
+  {name, version}: ConfigArgs = {},
+  provider: TracerProvider,
+): TracerRef => {
+  // otel
+  const tracerRef = useRef<Tracer | null>(null);
+
+  // using the layout effect to make sure the tracer is initialized before the component is rendered
+  useEffect(() => {
+    if (tracerRef.current === null && provider) {
+      trace.setGlobalTracerProvider(provider);
+
+      tracerRef.current = trace.getTracer(
+        name || TRACER_DEFAULT.name,
```

Comment:
given 'name' is required in `getTracer` I'd maybe make that required in this hook as well and then maybe just omit a version when it isn't set since '1.0' may not always be reasonable as a default